src/config.rs

```diff
@@ -414,6 +471,101 @@ impl Config {
     pub fn mirror_map(&self) -> &std::collections::HashMap<Url, Vec<Url>> {
         &self.mirrors
     }
+
+    /// Modify this config with the given key and value
+    ///
+    /// # Note
+    ///
+    /// It is required to call `save()` to persist the changes.
+    pub fn set(&mut self, key: &str, value: Option<String>) -> miette::Result<()> {
+        match key {
+            "default-channels" => {
+                self.default_channels = value
+                    .map(|v| serde_json::de::from_str(&v))
+                    .transpose()
+                    .into_diagnostic()?
+                    .unwrap_or_default();
+            }
+            "change-ps1" => {
+                self.change_ps1 = value.map(|v| v.parse()).transpose().into_diagnostic()?;
+            }
+            "tls-no-verify" => {
+                self.tls_no_verify = value.map(|v| v.parse()).transpose().into_diagnostic()?;
+            }
+            "repodata-config.disable-jlap" => {
+                self.repodata_config
+                    .get_or_insert(RepodataConfig::default())
+                    .disable_jlap = value.map(|v| v.parse()).transpose().into_diagnostic()?;
+            }
+            "repodata-config.disable-bzip2" => {
+                self.repodata_config
+                    .get_or_insert(RepodataConfig::default())
+                    .disable_bzip2 = value.map(|v| v.parse()).transpose().into_diagnostic()?;
+            }
+            "repodata-config.disable-zstd" => {
+                self.repodata_config
+                    .get_or_insert(RepodataConfig::default())
+                    .disable_zstd = value.map(|v| v.parse()).transpose().into_diagnostic()?;
+            }
+            "pypi-config.index-url" => {
+                self.pypi_config.index_url = value
+                    .map(|v| Url::parse(&v))
+                    .transpose()
+                    .into_diagnostic()?;
+            }
+            "pypi-config.keyring-provider" => {
+                self.pypi_config.keyring_provider = value
+                    .map(|v| match v.as_str() {
+                        "disabled" => Ok(KeyringProvider::Disabled),
+                        "subprocess" => Ok(KeyringProvider::Subprocess),
+                        _ => Err(miette::miette!("invalid keyring provider")),
+                    })
+                    .transpose()?;
+            }
+            _ => return Err(miette::miette!("unsupported key: {}", key)),
```

Comment:
I think it would be really nice if the user would get a list of options printed, I personally was making a stupid typo and it made me look into the code how it should be. 

Also do you have any idea's how we could test if we support all options?